### ISSUE TITLE
Regression: padding on skills window overextended

### ISSUE THREAD

**Issue by themightychris** [SlateFoundation/slate-cbl#420]
The lower portion of this window is supposed to be flush with the edges, I think this used to be accomplished with a negative margin

<img width="752" alt="screen shot 2018-01-26 at 7 21 44 pm" src="https://user-images.githubusercontent.com/458494/35466103-00bb2932-02cf-11e8-838b-d4aafcade6e2.png">

**Comment by themightychris** [comment]
One thing I've been trying to do in this rewrite is to implement major views as Panels and detach all their logic and styling from whether or not they're shown in a window. The goal being to make our major pieces more configurable and reusable building blocks, and isolate the putting of them into windows as a usage detail.

This creates some challenges though, like in this case we might need to apply some special styling to parts of this panel only when it's in a window. The window styling shouldn't have special cases for this particular component, and the component shouldn't be styled under an assumption that it's in a window. I just applied this quick fix for this case which achieves being able to use the component in or out of a window, but I don't like that it leaves the SkillList component having to "know" that windows apply 16px of padding, and doesn't address other potential padded containers: 04edeaccdc83ad39c4ccff588a4251051854c9c8

I'm thinking that the ideal solution should be a policy like:

- Reusable components should not come with any outer padding, that should be up to whatever container they're put in or set via bodyPadding config where the component is used
- There should be some standard semantic CSS class like `flush-with-container` for elements within a reusable component that are designed to be flush with the edges of any container border
- Our Window base class and its styling should then come set up to pad all contents except things with the flush class

I think this pattern would both address the window-container use case and a few other cases I've encountered where we had kludged in negative margin

@ryon thoughts?
